apps/e2e/ag-grid-storybook-e2e/cypress.config.ts

```diff
@@ -1,21 +1,85 @@
 import { nxE2EPreset } from '@nrwl/cypress/plugins/cypress-preset';
+import * as utils from '@percy/sdk-utils';
 
 import { defineConfig } from 'cypress';
+import * as fs from 'fs';
 
-const cypressJsonConfig = {
-  fileServerFolder: '.',
-  fixturesFolder: './src/fixtures',
-  video: false,
-  videosFolder: '../../../dist/cypress/apps/e2e/ag-grid-storybook-e2e/videos',
-  screenshotsFolder:
-    '../../../dist/cypress/apps/e2e/ag-grid-storybook-e2e/screenshots',
-  chromeWebSecurity: false,
-  specPattern: 'src/e2e/**/*.cy.{js,jsx,ts,tsx}',
-  supportFile: 'src/support/e2e.ts',
-};
 export default defineConfig({
   e2e: {
     ...nxE2EPreset(__dirname),
-    ...cypressJsonConfig,
+    video: false,
+    setupNodeEvents: (on, config) => {
+      // Use a large window size so screenshots fit within the viewport.
+      on('before:browser:launch', (browser, launchOptions) => {
+        const width = 1920;
+        const height = 1600;
+
+        if (browser.name === 'chrome' && browser.isHeadless) {
+          launchOptions.args.push(`--window-size=${width},${height}`);
+          launchOptions.args.push('--force-device-scale-factor=2');
+          launchOptions.args.push('--no-sandbox');
+          launchOptions.args.push('--disable-setuid-sandbox');
+        }
+
+        if (browser.name === 'electron' && browser.isHeadless) {
+          launchOptions.preferences.width = width;
+          launchOptions.preferences.height = height;
+          launchOptions.preferences.deviceScaleFactor = 2;
+        }
+
+        if (browser.name === 'firefox' && browser.isHeadless) {
+          launchOptions.args.push(`--window-size=${width},${height}`);
+        }
+
+        return launchOptions;
+      });
+
+      on(
+        'after:screenshot',
+        async (
+          details: Cypress.ScreenshotDetails
+        ): Promise<Cypress.AfterScreenshotReturnObject> => {
```

Comment:
This method hooks into Cypress's screenshot and hands that image off to Percy.

Reply:
This comment would be excellent to have in the source code, too.